homeassistant/components/wsdot/sensor.py

```diff
@@ -44,22 +55,67 @@
 async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
-    add_entities: AddEntitiesCallback,
+    async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Migrate a platform-style wsdot to entry-style."""
+    entries = hass.config_entries.async_entries(DOMAIN)
+    if entries:
+        for entry in entries:
+            if entry.data[CONF_API_KEY] in [
+                config_entry[CONF_API_KEY]
+                for config_entry in config[Platform.SENSOR]
+                if config_entry[CONF_PLATFORM] == DOMAIN
+            ]:
```

Comment:
instead, leave it to the config flow to abort if we already have this entry

Reply:
So you don't want an issue telling the user to clean up the configuration.yaml?

Reply:
Yes we do, but we should move the duplication check to the config flow instead

Reply:
Checkout https://github.com/home-assistant/core/pull/151567/files#diff-b50e867087c6d391d05b571f7c5870ea294958433e507c4e2629466a3496d914R59-R105